commands/delete-repository.ts

```diff
@@ -0,0 +1,167 @@
+        value: 'No',
+        prompt: 'Are you sure you want to delete this repository and its associated images? Enter Yes or No: '
+    };
+    let answer = await vscode.window.showInputBox(opt);
+    if (answer !== 'Yes') return;
+
+    // generate credentials before requesting a delete.
+    if (context) {
+        username = context.userName;
+        password = context.password;
+        repoName = context.label;
+        subscription = context.subscription;
+        registry = context.registry;
+    }
+    else { //this is separated from !context above so it only calls loginCredentials once user has assured they want to delete the repository
+        let creds = await loginCredentials(subscription, registry);
+        username = creds.username;
+        password = creds.password;
+    }
+    let path = `/v2/_acr/${repoName}/repository`;
+    let r = await request_data_from_registry('delete', registry.loginServer, path, username, password);
+}
+
+
+/**
+ *
+ * @param http_method : the http method, this function currently only uses delete
+ * @param login_server: the login server of the registry
+ * @param path : the URL path
+ * @param username : registry username, can be in generic form of 0's, used to generate authorization header
+ * @param password : registry password, can be in form of accessToken, used to generate authorization header
+ */
+async function request_data_from_registry(http_method: string, login_server: string, path: string, username: string, password: string) {
+    let url: string = `https://${login_server}${path}`;
+    let header = _get_authorization_header(username, password);
+    let opt = {
+        headers: { 'Authorization': header },
+        http_method: http_method,
+        url: url
+    }
+    let err = false;
+    try {
+        let response = await request.delete(opt);
+    } catch (error) {
+        err = true;
+        console.log(error);
+    }
+    if (!err) {
+        vscode.window.showInformationMessage('Successfully deleted repository');
+    }
+}
+
+/**
+ *
+ * @param username : username for creating header
+ * @param password : password for creating header
+ */
+function _get_authorization_header(username: string, password: string): string {
+    let auth = ('Basic ' + (encode(username + ':' + password).trim()));
+    return (auth);
+}
+
+
+//Implements new Service principal model for ACR container registries while maintaining old admin enabled use
+/**
+ * this function implements a new Service principal model for ACR and gets the valid login credentials to make an API call
+ * @param subscription : the subscription the registry is on
+ * @param registry : the registry to get login credentials for
+ * @param context : if command is invoked through a right click on an AzureRepositoryNode. This context has a password and username
+ */
+async function loginCredentials(subscription, registry, context?: AzureRepositoryNode): Promise<{ password: string, username: string }> {
```

Comment:
>loginCredentials [](start = 15, length = 16)

This is quite a reusable function. Why not move it to a acr-credential-manager.ts or global location?

Reply:
in delete Image I moved it to AzureCredentialsManager.ts is that a good location?

Reply:
Created a new file for these new functions, see updated commit